File: src/Form_NewChat.vb
Imports System
Imports System.Linq
Imports System.Windows.Forms

Namespace TmCGPTD
    Public Partial Class Form_NewChat
        Public ReadOnly Property EnteredText As String
            Get
                Return TextBox1.Text
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub ButtonOK_Click(ByVal sender As Object, ByVal e As EventArgs)
            DialogResult = DialogResult.OK
            Close()
        End Sub

        Private Sub ButtonCancel_Click(ByVal sender As Object, ByVal e As EventArgs)
            DialogResult = DialogResult.Cancel
            Close()
        End Sub

        Private Sub Form_NewChat_Shown(ByVal sender As Object, ByVal e As EventArgs)
            KeyPreview = True ' KeyPreview プロパティを True に設定
        End Sub

        Private Sub TextBox1_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs)
            If e.KeyCode = Keys.Return Then
                Dim controlStr = Controls.Find($"ButtonOK", True)
                Dim buttonStr As Button = TryCast(controlStr.FirstOrDefault(), Button) ' ボタンを取得
                buttonStr.PerformClick()
            End If
        End Sub
    End Class
End Namespace

File: src/Form_NewChat.cs
using System;
using System.Linq;
using System.Windows.Forms;

namespace TmCGPTD
{
    public partial class Form_NewChat
    {
        public string EnteredText
        {
            get
            {
                return TextBox1.Text;
            }
        }

        public Form_NewChat()
        {
            InitializeComponent();
        }

        private void ButtonOK_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
            Close();
        }

        private void ButtonCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void Form_NewChat_Shown(object sender, EventArgs e)
        {
            KeyPreview = true; // KeyPreview プロパティを True に設定
        }

        private void TextBox1_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Return)
            {
                Control[] controlStr = Controls.Find($"ButtonOK", true);
                Button buttonStr = controlStr.FirstOrDefault() as Button; // ボタンを取得
                buttonStr.PerformClick();
            }
        }
    }
}